pyne/mesh.py

```diff
@@ -775,11 +775,6 @@ def __init__(self, mesh=None, structured=False,
                 setattr(self, name, MaterialMethodTag(mesh=self, name=name,
                         doc=doc))
 
-        # add a property: cell_cell_fracs_tag and set it to None
-        self.cell_number_tag = None
-        self.cell_fracs_tag = None
-        self.cell_largest_frac_number_tag = None
-        self.cell_largest_frac_tag = None
```

Comment:
I am not sure that it makes sense to delete these, per se.

Reply:
The `setattr()` was called at the end of function `self.tag()`. Therefore, `self.tag()` adds a new tag to the mesh. So we don't need to define these four property members in class Mesh. They will be created when we need to use these data, i.e., when the function `tag_cell_fracs()` was called.

What's more, if I want to use the `self.tag()` to create these tags, I have to delete them. Because `self.tag()` won't create a tag if the tag name is exist.

PS: These changes have passed the test functions in tests/test_mesh.py. As you can see, I didn't change the test_mesh.py in this pull request. They give the identical result as previous method for tagging cell_fracs.

Reply:
@zxkjack123 I'm not sure that you *have* to delete them, because it seems that new tags aren't actually registered as part of the `self.tags` dictionary attribute anywhere in the `mesh.tag` method so this wouldn't be a problem. Did you actually get this error when trying it?

At any rate, until the `tag_cell_fracs` method is called, I don't see a reason to keep them around. Did you have a specific purpose for them being around but set to `None` @scopatz? Their existence can be checked for using the `hasattr` built-in if they're needed, and if they don't exist then your new method can be called to create them.

Reply:
@pshriwise Actually, I haven't tried to call updated `tag_cell_fracs` with the `self.cell_number_tag`, .... And these tag names are not registered as part of the `self.tags`. I mistook these tag names with the `self.tags`. 

I agree with you that these tags should not be around but set to `None`, as they are necessary only for application of sub-voxel r2s.